### PR TITLE
【Hackathon No.89】 Remove circle import Part2

### PR DESCRIPTION
**Pull request by Liyulingyue**
### PR types

Others 
### PR changes

Others 
### Describe

- https://github.com/PaddlePaddle/Paddle/issues/50663#task89

### Details
- 根据https://github.com/PaddlePaddle/Paddle/discussions/50711 将jit下的import移动到文件头部
- python\paddle\jit\dy2static\call_transformer.py
    - from paddle.jit.dy2static.convert_call_func import is_builtin 直接移动到文件头
- python\paddle\jit\dy2static\convert_call_func.py
    - import six：PASS
    - from paddle.nn import Sequential：未处理
    - from paddle.jit.dy2static.program_translator import (
        StaticFunction,
        convert_to_static,
        unwrap_decorators,
    )：移动CONVERSION_OPTIONS的声明到program_translator，统一import program_translator
- python\paddle\jit\dy2static\convert_operators.py
    - from paddle.fluid.dygraph.base import in_declarative_mode：暂不处理
    - from paddle.static.nn.control_flow import Assert：暂不处理
- python\paddle\jit\dy2static\partial_program.py
    - from paddle.incubate.autograd.primapi import to_prim：暂不处理
    - from paddle.amp.auto_cast import _in_amp_guard, _in_pure_fp16_guard：暂不处理
- python\paddle\jit\dy2static\program_translator.py
    - from paddle.static import InputSpec：暂不处理
    - from paddle.incubate.autograd.primapi import to_prim：暂不处理
    - from paddle.fluid.dygraph.base import _switch_declarative_mode_guard_：暂不处理
    - from paddle.jit.dy2static.program_translator import (
                        ProgramTranslator,
                    )：直接删除
- python\paddle\jit\dy2static\utils.py
    - from paddle.jit.dy2static.return_transformer import (
        RETURN_NO_VALUE_MAGIC_NUM,
    )：将声明迁移到utils
    - import paddle  # noqa: F401
    - import paddle.fluid as fluid  # noqa: F401
    - import paddle.fluid.dygraph as dygraph  # noqa: F401
    - import paddle.fluid.layers as layers  # noqa: F401
    - import paddle.jit.dy2static as _jst  # noqa: F401
    - from paddle import to_tensor  # noqa: F401
    - from paddle.fluid.dygraph import to_variable  # noqa: F401
    - import numpy as np  # noqa: F401：上述均暂不修改
    - from .static_analysis import StaticAnalysisVisitor：暂不处理，在下个PR中解决
    - from paddle.jit.dy2static.static_analysis import NodeVarType：暂不处理，在下个PR中解决，这两个对static_analysis的引用可能都需要迁移logging和static中的实现，可能需要修改_init_.py
    - from paddle.jit.dy2static.ifelse_transformer import (
        FALSE_FUNC_PREFIX,
        TRUE_FUNC_PREFIX,
    )：将声明移动到utils
    - from paddle.jit.dy2static.loop_transformer import (
        FOR_BODY_PREFIX,
        FOR_CONDITION_PREFIX,
        WHILE_BODY_PREFIX,
    )： 将声明移动到utils

### CODE REVIEW

File: python/paddle/jit/dy2static/convert_call_func.py [hunk 1]
```diff
@@ -12,22 +12,20 @@
...
 import re
-import types
 from typing import Any, List
 
 import numpy
 
 from paddle.fluid.dygraph.layers import Layer
 from paddle.jit.dy2static.logging_utils import TranslatorLogger
-from paddle.jit.dy2static.utils import is_paddle_func, unwrap
+from paddle.jit.dy2static.utils import is_builtin, is_paddle_func, unwrap
```
Comment (Aurelius84) [inline python/paddle/jit/dy2static/convert_call_func.py]: 后续 `from paddle.jit.dy2static.xxx`  可以考虑优化为 `from .xxx` 么？

File: python/paddle/jit/dy2static/return_transformer.py [hunk 2]
```diff
@@ -44,7 +44,6 @@
 # should return.
 
 # Assign not support float64, use float32 value as magic number.
-RETURN_NO_VALUE_MAGIC_NUM = 1.77113e27
 RETURN_NO_VALUE_VAR_NAME = "__no_value_return_var"
```
Comment (Aurelius84) [inline python/paddle/jit/dy2static/return_transformer.py]: RETURN_NO_VALUE_VAR_NAME 是否也可以考虑到utils.py里，与RETURN_NO_VALUE_MAGIC_NUM这个magic number放一起？

File: python/paddle/jit/dy2static/utils.py [hunk 3]
```diff
@@ -1109,6 +1110,14 @@ def merge_from(self, name_scope):
         self.push_pop_vars |= name_scope.push_pop_vars
 
 
+TRUE_FUNC_PREFIX = 'true_fn'
+FALSE_FUNC_PREFIX = 'false_fn'
+
+WHILE_BODY_PREFIX = 'while_body'
+FOR_CONDITION_PREFIX = 'for_loop_condition'
```
Comment (Aurelius84) [inline python/paddle/jit/dy2static/utils.py]: 可否将这些全局的config字段统一放到文件import 语句后面？

**Comment by Liyulingyue** [comment]
@Aurelius84 根据上述的建议修改了本PR涉及的内容。codestyle那块没法过去，因为对应的包是通过执行字符串的形式使用的，所以必须保留引用~

File: python/paddle/jit/dy2static/call_transformer.py [hunk 4]
```diff
@@ -17,6 +17,7 @@
 from paddle.utils import gast
 
 from .base_transformer import BaseTransformer
+from .utils import is_builtin
```
Comment (Aurelius84) [inline python/paddle/jit/dy2static/call_transformer.py]: 可以改成这样，跳过import-but-unused的检查，参考：https://flake8.pycqa.org/en/2.6.0/warnings.html

```suggestion
from .utils import is_builtin   # noqa: F401
```

File: python/paddle/jit/dy2static/call_transformer.py [hunk 5]
```diff
@@ -17,6 +17,7 @@
 from paddle.utils import gast
 
 from .base_transformer import BaseTransformer
+from .utils import is_builtin   # noqa: F401
```
Comment (Liyulingyue) [inline python/paddle/jit/dy2static/call_transformer.py]:
```suggestion
from .utils import is_builtin  # noqa: F401
```

**Review by Aurelius84 — DISMISSED** [review]
LGTM

**Comment by luotao1** [comment]
@Liyulingyue 请解决下冲突

**Review by Aurelius84 — APPROVED** [review]
LGTM

**Pull request merged**